api/s2n.h

```diff
@@ -596,6 +596,269 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
 S2N_API 
 extern int s2n_cert_get_utf8_string_from_extension_data(const uint8_t *extension_data, uint32_t extension_len, uint8_t *out_data, uint32_t *out_len);
 
+/* Pre-shared key (PSK) Hash Algorithm - RFC 8446 Section-2.2 */
+/* https://tools.ietf.org/html/rfc8446#section-2.2 */
+typedef enum {
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk;
+
+/**
+ * Creates a new s2n external Pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default 
+ * PSK hash algorithm. An external PSK is an out-of-band provisioned PSK established 
+ * externally using a secure mutually agreed upon mechanism. 
+ * 
+ * # Safety 
+ *
+ * Use `s2n_psk_free` to free the memory allocated to the s2n external PSK object created by this API. 
+ *
+ * The external PSK authentication mechanism in >= TLS1.3 implicitly assumes each PSK is known to exactly one client 
+ * and one server, and these peers never switch roles. If this assumption is violated, the security properties of 
+ * >= TLS1.3 are severely weakened
```

Comment:
Is this the right place for this warning? It's not really related to creating a new PSK object. It might belong in the usage guide or on set-secret instead.